rows.go

```diff
@@ -65,11 +66,19 @@ func (rows *mysqlRows) Columns() []string {
 }
 
 func (rows *mysqlRows) Close() (err error) {
+	if f := rows.finish; f != nil {
```

Comment:
because of https://github.com/go-sql-driver/mysql/pull/608#discussion_r120251781 .

> `mc.startWatch` and `mc.finish()` must be paired, but `mysqlRows` is used internally.
> 
> - getWarnings: https://github.com/go-sql-driver/mysql/blob/v1.3/errors.go#L92
> 
> So, I use `setfinish` to prevent `mc.finish()` from being called unexpectedly.

So, rows generated by `QueryContext` **MUST** call `mc.finish()` in `Close()`, but rows generated by `Query` **MUST NOT** call `mc.finish()`.